### DISCUSSION TITLE
Suggestion: Add a link to the git repo

### DISCUSSION THREAD

**Discussion by simonmysun** [Crissium/SilverDict#14] [Ideas]
I would suggest adding a link to the git repo in the front end. 

A possible approach is to add a new entry in the menu bar[[1]] or in the settings tab[[2]]. It should include the name of the author, the link to the repo and the link to the issue page. Another possilbe place to add is at the bottom of the query page[[3]]. 

[1]: https://github.com/Crissium/SilverDict/blob/9e55ccb27980138f0b235454160ece4d3fc9aec7/client/src/App.js#L28
[2]: https://github.com/Crissium/SilverDict/blob/9e55ccb27980138f0b235454160ece4d3fc9aec7/client/src/components/SettingsScreen.js#L147
[3]: https://github.com/Crissium/SilverDict/blob/9e55ccb27980138f0b235454160ece4d3fc9aec7/client/src/components/QueryScreen.js#L35

This will encourage collaboration and channel the user who wants to complain or replicate to the right place. The backlinks also improve the reputy and search ranking of this repo. 

P.S. regarding search ranking, I would also suggest adding `<meta>` tags to improve machine readability and attribution. [Here](https://gist.github.com/lancejpollard/1978404)'s a list of useful `<meta>` tags. I believe at least `keywords` and `description` is necessary. 

How do you think?

**Comment by Crissium** [comment]
First of all thanks for your continuing interest in this project!

This is certainly a very good idea because the majority of applications have an 'About' drop-down menu option/a tab in the settings dialogue/etc. In addition to author/source code repo/issue tracker I think we could also include the version.

> A possible approach is to add a new entry in the menu bar[[1](https://github.com/Crissium/SilverDict/blob/9e55ccb27980138f0b235454160ece4d3fc9aec7/client/src/App.js#L28)] or in the settings tab[[2](https://github.com/Crissium/SilverDict/blob/9e55ccb27980138f0b235454160ece4d3fc9aec7/client/src/components/SettingsScreen.js#L147)]. It should include the name of the author, the link to the repo and the link to the issue page. Another possilbe place to add is at the bottom of the query page[[3](https://github.com/Crissium/SilverDict/blob/9e55ccb27980138f0b235454160ece4d3fc9aec7/client/src/components/QueryScreen.js#L35)].

Good idea! Another place to try is a dedicated 'About' screen, which many mobile apps have now.

> P.S. regarding search ranking, I would also suggest adding <meta> tags to improve machine readability and attribution. [Here](https://gist.github.com/lancejpollard/1978404)'s a list of useful <meta> tags. I believe at least keywords and description is necessary.

Yes, I fully agree. In fact initially when the front-end still used the deprecated create-react-app, there were such <meta> tags in `index.html`. After I migrated to vite, which doesn't have them in the template, I forgot to add them back.

Actually I have some changes locally that are still uncommitted and I am very busy at the moment. Hopefully I'll come back to this project at the start of August and make all the changes mentioned above. Thanks again!

**Comment by Crissium** [comment]
Done in commits https://github.com/Crissium/SilverDict/commit/52493b6a177999446860027c695f453e8fd615f5 and https://github.com/Crissium/SilverDict/commit/42a6d93e6b36de38966d7e621e1f9171864e4973.
